examples/gno.land/p/gnoland/commondao/record.gno

```diff
@@ -0,0 +1,189 @@
+		Choice VoteChoice
+
+		// Reason contains the reason for the vote.
+		Reason string
+
+		// Context can store any custom voting values related to the vote.
+		Context any
+	}
+
+	// ReadOnlyVotingRecord defines an interface for read only voting records.
+	ReadOnlyVotingRecord interface {
+		// Size returns the total number of votes that record contains.
+		Size() int
+
+		// VoteChoices returns the voting choices that has been voted.
+		VoteChoices() []VoteChoice
+
+		// Iterate iterates voting record votes.
+		Iterate(fn VoteIterFn) (stopped bool)
+
+		// VoteCount returns the number of votes for a single voting choice.
+		VoteCount(VoteChoice) int
+
+		// HasVoted checks if an account already voted.
+		HasVoted(std.Address) bool
+
+		// GetProvableMajorityChoice returns the choice voted by the majority.
+		// The result is only valid if there is a majority.
+		// Caller must validate that the returned choice represents a majority.
+		GetProvableMajorityChoice() VoteChoice
+	}
+)
+
+// VotingRecord stores accounts that voted and vote choices.
+type VotingRecord struct {
+	votes avl.Tree // string(address) -> Vote
+	count avl.Tree // string(choice) -> int
+}
+
+// Size returns the total number of votes that record contains.
+func (r VotingRecord) Size() int {
+	return r.votes.Size()
+}
+
+// VoteChoices returns the voting choices that has been voted.
+func (r VotingRecord) VoteChoices() []VoteChoice {
+	var choices []VoteChoice
+	r.count.Iterate("", "", func(k string, v any) bool {
+		choices = append(choices, VoteChoice(k))
+		return false
+	})
+	return choices
+}
+
+// Iterate iterates voting record votes.
+func (r VotingRecord) Iterate(fn VoteIterFn) bool {
+	return r.votes.Iterate("", "", func(k string, v any) bool {
+		return fn(v.(Vote))
+	})
+}
+
+// VoteCount returns the number of votes for a single voting choice.
+func (r VotingRecord) VoteCount(c VoteChoice) int {
+	if v, found := r.count.Get(string(c)); found {
+		return v.(int)
+	}
+	return 0
+}
+
+// HasVoted checks if an account already voted.
+func (r VotingRecord) HasVoted(user std.Address) bool {
+	return r.votes.Has(user.String())
+}
+
+// AddVote adds a vote to the voting record.
+// If a vote for the same user already exists is overwritten.
+func (r *VotingRecord) AddVote(vote Vote) (updated bool) {
+	// Get previous member vote if it exists
+	v, _ := r.votes.Get(vote.Address.String())
+
+	// When a previous vote exists update counter for the previous choice
+	updated = r.votes.Set(vote.Address.String(), vote)
+	if updated {
+		prev := v.(Vote)
+		r.count.Set(string(prev.Choice), r.VoteCount(prev.Choice)-1)
+	}
+
+	r.count.Set(string(vote.Choice), r.VoteCount(vote.Choice)+1)
+	return
+}
+
+// GetProvableMajorityChoice returns the choice voted by the majority.
+// The result is only valid if there is a majority.
+// Caller must validate that the returned choice represents a majority.
+func (r VotingRecord) GetProvableMajorityChoice() VoteChoice {
```

Comment:
Planning to rethink this functionality

Reply:
Addressed in 68689be.

Now it's not part of the voting record but instead is a new `FindMostVotedChoice()` function